### PR TITLE
add `integrity_check` PRAGMA

### PR DESCRIPTION
**Pull request by mishal23**
Fixes: #731

### CODE REVIEW

**Comment by fnc12** [comment]
@mishal23 thank you for submitting this PR. Please answer one question I left at review section

File: dev/pragma.h [hunk 1]
```diff
@@ -65,6 +65,16 @@ namespace sqlite_orm {
             this->set_pragma("auto_vacuum", value);
         }
 
+        std::string integrity_check() {
+            return this->get_pragma<std::string>("integrity_check");
+        }
+
+        std::string integrity_check(std::string table_name) {
```
[the remark this answers is not in the document]
Comment (mishal23) [reply dev/pragma.h:72]: sure, that makes sense, I'll modify it and also fix the clang-linter issue in the next commit.

[the remark this answers is not in the document]
Comment (mishal23) [reply dev/pragma.h:72]: So we were thinking about it and if we change to a template, it'll be open for erroneous usage with different data types as well, so we were thinking of doing template type checking in the code, as shown below:

```cpp
        template<class T>
        std::string integrity_check(T table_name) {
            if(std::is_same<std::string, T>::value || std::is_same<std::string_view, T>::value) {
                std::ostringstream oss;
                oss << "integrity_check(" << table_name << ")";
                return this->get_pragma<std::string>(oss.str());
            }
            else {
                return "invalid type, only string/string_view is supported";
            }
        }
```

but the issue is there are multiple string types as seen here: https://en.cppreference.com/w/cpp/string

We can probably have no checks or checks with only `string` & `string_view`. do you have any recommendations on what we should do?

[the remark this answers is not in the document]
Comment (fnc12) [reply dev/pragma.h:72]: no. This check will disable custom string usage (e.g. `QString` and `Glib::ustring`). The case is that argument must be serializable to ostream and that is all

[the remark this answers is not in the document]
Comment (mishal23) [reply dev/pragma.h]: Ok makes sense, added the template argument for `integrity_check`

**Comment by fnc12** [comment]
@mishal23 looks nice. Let's wait appveyor build and then I merge it

**Comment by mishal23** [comment]
Hey @fnc12, the build doesn't seem to be running, is there anything which is needed to be done?

**Comment by fnc12** [comment]
IDK what happened. Anyway everything was build successfully on another check so let's merge as is.

**Comment by fnc12** [comment]
@mishal23 one thing: please head this PR to `dev`. All new features must be merged into `dev` not `master`

**Comment by mishal23** [comment]
Changing the base branch to merge doesn't look good, some other commits are also getting included, I'll open a new PR after making a new branch from dev as the source

**Comment by fnc12** [comment]
@mishal23 ok

**Comment by mishal23** [comment]
closing this PR, and opened #734

**Pull request closed without merging**